app/src/main/java/com/keylesspalace/tusky/components/instanceinfo/InstanceInfoRepository.kt

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2022 Tusky contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.instanceinfo
+
+import android.util.Log
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.db.AppDatabase
+import com.keylesspalace.tusky.db.EmojisEntity
+import com.keylesspalace.tusky.db.InstanceInfoEntity
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.VersionUtils
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class InstanceInfoRepository @Inject constructor(
+    private val api: MastodonApi,
+    db: AppDatabase,
+    accountManager: AccountManager
+) {
+
+    private val dao = db.instanceDao()
```

Comment:
We use dao only as a fallback but we could also have some cache interval in which we don't re-load it again. It almost never changes anyway. But this is for another day I think.